flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlAlterTable.java

```diff
@@ -36,40 +36,61 @@
 import static java.util.Objects.requireNonNull;
 
 /**
- * Abstract class to describe statements like ALTER TABLE [[catalogName.] dataBasesName].tableName
- * ...
+ * Abstract class to describe statements like ALTER TABLE IF EXISTS [[catalogName.]
+ * dataBasesName].tableName ...
  */
 public abstract class SqlAlterTable extends SqlCall {
 
     public static final SqlSpecialOperator OPERATOR =
             new SqlSpecialOperator("ALTER TABLE", SqlKind.ALTER_TABLE);
 
+    protected final boolean ifExists;
     protected final SqlIdentifier tableIdentifier;
     protected final SqlNodeList partitionSpec;
 
+    public SqlAlterTable(SqlParserPos pos, SqlIdentifier tableName) {
+        this(pos, false, tableName);
+    }
+
+    public SqlAlterTable(SqlParserPos pos, boolean ifExists, SqlIdentifier tableName) {
```

Comment:
The order (SqlParserPos, SqlIdentifier, boolean) would feel more natural here, I think, since the table identifier is more important than the flag. The same also for the other constructor which has the boolean.

In the DDL the boolean comes first, but in the Java code I think it is more logical to have optional flags in the end.